connector/src/main/java/org/geysermc/connector/utils/AdvancementsUtils.java

```diff
@@ -0,0 +1,147 @@
+        SimpleFormWindow menuForm = (SimpleFormWindow) session.getWindowCache().getWindows().get(ADVANCEMENTS_MENU_FORM_ID);
+        menuForm.setResponse(response);
+        SimpleFormResponse formResponse = (SimpleFormResponse) menuForm.getResponse();
+        int clickedButton = -1;
+
+        if (formResponse != null && formResponse.getClickedButton() != null) {
+            clickedButton = formResponse.getClickedButtonId();
+
+        }
+        StringBuilder content = new StringBuilder();
+        String[] idAndTitle = session.getButtonIdsToIdAndTitleAdvancements().get(clickedButton);
+
+        content.append(LanguageUtils.getPlayerLocaleString("geyser.advancements.earned", language) + ":\n");
+        try {
+            for (Map.Entry<String, Advancement> advancementEntry : session.getStoredAdvancements().entrySet()) {
+
+                if (advancementEntry.getValue().getId().startsWith(idAndTitle[0].replace("/root", "").replace("root", ""))) {
+                    content.append(MessageTranslator.convertMessage(advancementEntry.getValue().getDisplayData().getTitle().toString()) + "\n");
+                }
+            }
+
+            // Showed advancements you have earned based on category selected
+            SimpleFormWindow window = new SimpleFormWindow(MessageTranslator.convertMessage(idAndTitle[1]), content.toString());
+            for (Map.Entry<String, Advancement> advancement : session.getStoredAdvancements().entrySet()) {
+                if (advancement.getValue().getId().startsWith(idAndTitle[0].replace("/root", ""))) {
+                    content.append(" - " + MessageTranslator.convertMessage(advancement.getValue().getDisplayData().getTitle().toString()) + "\n     " + MessageTranslator.convertMessage(advancement.getValue().getDisplayData().getDescription().toString()) + "\n");
+                }
+            }
+            window.getButtons().add(new FormButton(LocaleUtils.getLocaleString("gui.back", language)));
+
+
+            session.sendForm(window, ADVANCEMENTS_LIST_FORM_ID);
+        } catch (NullPointerException e) {
+        }
+        return true;
+    }
+
+    /**
+     * Handle the list form response (back button)
+     *
+     * @param session  The session that sent the response
+     * @param response The response string to parse
+     * @return True if the form was parsed correctly, false if not
+     */
+    public static boolean handleListForm(GeyserSession session, String response) {
+        SimpleFormWindow listForm = (SimpleFormWindow) session.getWindowCache().getWindows().get(ADVANCEMENTS_LIST_FORM_ID);
+        listForm.setResponse(response);
+        SimpleFormResponse formResponse = (SimpleFormResponse) listForm.getResponse();
+        if (!listForm.isClosed() && formResponse != null && formResponse.getClickedButton() != null) {
+            session.sendForm(AdvancementsUtils.buildMenuForm(session), ADVANCEMENTS_MENU_FORM_ID);
```

Comment:
You don't need to reference the current class inside of itself, you can just call `buildMenuForm`.